File: 0038-count-and-say/0038-count-and-say.py
from collections import Counter
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1:
            return '1'
        ans='1'
        for i in range(n-1):
            new_ans=""
            i=0
            while i<len(ans):
                count=1
                while i+1<len(ans) and ans[i]==ans[i+1]:
                    count+=1
                    i+=1
                new_ans+=str(count)+ans[i]
                i+=1
            ans=new_ans
        return ans

        

File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string res="1";
        for(int i=0;i<n-1;i++){
            string temp="";
            int j=0;
            while(j<res.size()){
                int c=1;
                while((j+1<res.size())&&(res[j]==res[j+1])){
                    c++;
                    j++;
                }
                temp+=to_string(c)+res[j];
                j++;
            }
            res=temp;
        }
        return res;
    }
};  